pilot/pkg/security/authz/model/util.go

```diff
@@ -27,14 +27,14 @@ import (
 // 4. Suffix match. For example, "*/review" matches "/bookstore/review", "/products/review", etc.
 func stringMatch(a string, list []string) bool {
 	for _, s := range list {
-		if a == s || s == "*" || prefixMatch(a, s) || suffixMatch(a, s) {
+		if a == s || s == "*" || prefixMatch(a, s) || prefixMatch(s, a) || suffixMatch(a, s) || suffixMatch(s, a) {
```

Comment:
Why change this?

Reply:
If you look at the tests, you can see that previously the pattern can only exist in the list. I extended to allow pattern can exist in `a` too. So now it supports both cases:
`S: "*local", List: []string{"cluster.local"}` and `S: "cluster.local", List: []string{"*local"}`. I can write another function to extend this and use it, or use the existing one.

Reply:
I know what the changes you made here I'm asking why make the change in this PR?

The change will affect other place: https://github.com/istio/istio/blob/96109c83af15b9c431aef6428127e606403c9d57/pilot/pkg/security/authz/model/permission.go#L52 so it's changing the API of the policy.

Reply:
I made the change because I thought it wouldn't affect the current implementation, because `service.Name` cannot have prefix/suffix if I understand it correctly? But I'll create a different function for this PR.

Reply:
I suggest not to change this and depend on external behavior. You could create a separate function or even just embed it in your main function as it's only used by your code.

Reply:
Will do. Thanks!